homeassistant/components/comelit/config_flow.py

```diff
@@ -0,0 +1,140 @@
+"""Config flow for Comelit integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from aiocomelit import ComeliteSerialBridgeAPi, exceptions as aiocomelit_exceptions
+import voluptuous as vol
+
+from homeassistant import core, exceptions
+from homeassistant.config_entries import ConfigEntry, ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PIN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import _LOGGER, DEFAULT_HOST, DEFAULT_PIN, DOMAIN
+
+
+def user_form_schema(user_input: dict[str, Any] | None) -> vol.Schema:
+    """Return user form schema."""
+    user_input = user_input or {}
+    return vol.Schema(
+        {
+            vol.Optional(CONF_HOST, default=DEFAULT_HOST): str,
+            vol.Optional(CONF_PIN, default=DEFAULT_PIN): str,
+        }
+    )
+
+
+STEP_REAUTH_DATA_SCHEMA = vol.Schema({vol.Required(CONF_PIN): str})
+
+
+async def validate_input(
+    hass: core.HomeAssistant, data: dict[str, Any]
+) -> dict[str, str]:
+    """Validate the user input allows us to connect."""
+
+    api = ComeliteSerialBridgeAPi(data[CONF_HOST], data[CONF_PIN])
+
+    try:
+        await api.login()
+    except aiocomelit_exceptions.CannotConnect as err:
+        raise CannotConnect from err
+    except aiocomelit_exceptions.CannotAuthenticate as err:
+        raise InvalidAuth from err
+    finally:
+        await api.logout()
+        await api.close()
+
+    return {"title": data[CONF_HOST]}
+
+
+class ComelitConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Comelit."""
+
+    VERSION = 1
+    entry: ConfigEntry | None = None
```

Comment:
```suggestion
    _reauth_entry: ConfigEntry
    _reauth_host: str
```

Reply:
To make mypy happy:

`_reauth_entry: ConfigEntry | None`